### DISCUSSION TITLE
Reason for not using WebGL

### DISCUSSION THREAD

**Discussion by alabhyajindal** [reaviz/reaflow#208] [General]
Hey!

I was exploring different flow libraries for a project and noticed reaflow and [reagraph](https://github.com/reaviz/reagraph). I noticed that reagraph is made using WebGL but reaflow is not. What was the reason behind this decision? Do you think that using WebGL for making flow based programming libraries is not a good idea?

**Comment by amcdnl** [comment]
Great question - Let me take a stab at answering...

WebGL is a very powerful technology but it comes with its own advantages and challenges. Where WebGL really shines is for very large visualizations like network graphs which is why reagraph implements it. It can be difficult to work with simple things ( curved lines for example ) and can have a large memory footprint.

Reaflow on the other hand is more designed for flow charts which don't typically have thousands of links and nodes. Since its rendered in SVG, the memory footprint is much smaller and you don't have all the challenges that something like WebGL has.

**Comment by alabhyajindal** [comment]
Thanks!

I understand that Reaflow is made for flow charts which typically don't have thousands of nodes and edges. But what about the use case of flow charts which _do_ have thousands of nodes and edges. What would you recommend for that? Is making a library tailored for this using [three.js](https://threejs.org/) a good idea?
